pkg/infra/aws/aws.go

```diff
@@ -13,3 +13,270 @@
+	})
+	if err != nil {
+		return err
+	}
+
+	_, err = ec2.NewDefaultRouteTable(ctx, "mo-pulumi-routetable", &ec2.DefaultRouteTableArgs{
+		DefaultRouteTableId: vpc.DefaultRouteTableId,
+		Routes: ec2.DefaultRouteTableRouteArray{
+			ec2.DefaultRouteTableRouteInput(&ec2.DefaultRouteTableRouteArgs{
+				CidrBlock: pulumi.String("0.0.0.0/0"),
+				GatewayId: igw.ID(),
+			}),
+		},
+	})
+	if err != nil {
+		return nil
+	}
+
+	eksRole, err := iam.NewRole(ctx, "eks-iam-eksRole", &iam.RoleArgs{
+		AssumeRolePolicy: pulumi.String(`{
+		    "Version": "2008-10-17",
+		    "Statement": [{
+		        "Sid": "",
+		        "Effect": "Allow",
+		        "Principal": {
+		            "Service": "eks.amazonaws.com"
+		        },
+		        "Action": "sts:AssumeRole"
+		    }]
+		}`),
+	})
+
+	if err != nil {
+		return err
+	}
+
+	eksPolicies := []string{
+		"arn:aws-cn:iam::aws:policy/AmazonEKSServicePolicy",
+		"arn:aws-cn:iam::aws:policy/AmazonEKSClusterPolicy",
+	}
+	for i, eksPolicy := range eksPolicies {
+		_, err := iam.NewRolePolicyAttachment(ctx, fmt.Sprintf("rpa-%d", i), &iam.RolePolicyAttachmentArgs{
+			PolicyArn: pulumi.String(eksPolicy),
+			Role:      eksRole.Name,
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	// Create the EC2 NodeGroup Role
+	nodeGroupRole, err := iam.NewRole(ctx, "nodegroup-iam-role", &iam.RoleArgs{
+		AssumeRolePolicy: pulumi.String(`{
+		    "Version": "2012-10-17",
+		    "Statement": [{
+		        "Sid": "",
+		        "Effect": "Allow",
+		        "Principal": {
+		            "Service": "ec2.amazonaws.com.cn"
+		        },
+		        "Action": "sts:AssumeRole"
+		    }]
+		}`),
+	})
+	if err != nil {
+		return err
+	}
+
+	nodeGroupPolicies := []string{
+		"arn:aws-cn:iam::aws:policy/AmazonEKSWorkerNodePolicy",
+		"arn:aws-cn:iam::aws:policy/AmazonEKS_CNI_Policy",
+		"arn:aws-cn:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly",
+	}
+	for i, nodeGroupPolicy := range nodeGroupPolicies {
+		_, err := iam.NewRolePolicyAttachment(ctx, fmt.Sprintf("ngpa-%d", i), &iam.RolePolicyAttachmentArgs{
+			Role:      nodeGroupRole.Name,
+			PolicyArn: pulumi.String(nodeGroupPolicy),
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	// Create a Security Group that we can use to actually connect to our cluster
+	sg, err := ec2.NewSecurityGroup(ctx, "mo-pulumi-sg", &ec2.SecurityGroupArgs{
+		Description: pulumi.String("security group for ec2 nodes"),
+		Name:        pulumi.String("mo-pulumi-sg"),
+		VpcId:       vpc.ID(),
+		Egress: ec2.SecurityGroupEgressArray{
+			ec2.SecurityGroupEgressArgs{
+				Protocol:   pulumi.String("-1"),
+				FromPort:   pulumi.Int(0),
+				ToPort:     pulumi.Int(0),
+				CidrBlocks: pulumi.StringArray{pulumi.String("0.0.0.0/0")},
+			},
+		},
+		Ingress: ec2.SecurityGroupIngressArray{
+			ec2.SecurityGroupIngressArgs{
+				Protocol:   pulumi.String("tcp"),
+				FromPort:   pulumi.Int(80),
+				ToPort:     pulumi.Int(80),
+				CidrBlocks: pulumi.StringArray{pulumi.String("0.0.0.0/0")},
```

Comment:
default to allowing all in- traffic?

Reply:
yeah, I change it to next commit.  more detail：https://docs.aws.amazon.com/zh_cn/vpc/latest/userguide/VPC_SecurityGroups.html